File: Problem Set 6 - Python/cash/cash.py
from cs50 import get_float

fint = 0
coins = 0

penny = 1
nickel = 5
dime = 10
quarter = 25

# get a float from the user
f = get_float("Change owed: ")
f = round(f * 100)

# keep subtracting a number till you cannot subtract it anymore, then move on to a smaller value
while f < 0:
    f = get_float("Change owed: ")
    f = round(f * 100)

while f - quarter >= 0:
    f = f - quarter
    coins += 1

while f - dime >= 0:
    f = f - dime
    coins += 1

while f - nickel >= 0:
    f = f - nickel
    coins += 1

while f - penny >= 0:
    f = f - penny
    coins += 1

# print the number of coins required
print(coins)

File: Problem Set 1 - Intro to C/Cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
 
int main (void)
{
   int fint;
   float f = 0;
   int coins = round(f * 10);
 
   float penny = 1;
   float nickel = 5;
   float dime = 10;
   float quarter = 25;
  
 
   do
   {
       f = get_float("Change owed: ");
       f = round(f * 100);
   }while (f < 0);
 
   while (f - quarter >= 0)
   {
       f = f - quarter;
       coins++;
   }
  
 
   while (f - dime >= 0)
   {
       f = f - dime;
       coins++;
   }
 
   while (f - nickel >= 0)
   {
       f = f - nickel;
       coins++;
   }
 
   while (f - penny >= 0)
   {
       f = f - penny;
       coins++;
   }
 
   //printf("%i\n", fint);
   printf("%i\n", coins);
 
  
 
 
 
}
 


